src/main/java/com/pub/wysio/service/CustomerService.java

```
@@ -10,33 +12,39 @@
 @Service
 public class CustomerService {
     @Autowired
-    private CustomerRepository customerRepository;
+    CustomerRepository customerRepository;
+    @Autowired
+    CustomerConverter customerConverter;
+
+    public CustomerDTO saveCustomer(CustomerDTO customerDTO) {
+        Customer customer = customerConverter.toEntity(customerDTO);
+        customerRepository.save(customer);
 
-    public Customer saveCustomer(Customer customer) {
-        return customerRepository.save(customer);
+        return customerConverter.toDto(customer);
     }
-    public List<Customer> getAllCustomers() {
-        return customerRepository.findAll();
+    public List<CustomerDTO> getAllCustomers() {
+        List<Customer> allCustomers = customerRepository.findAll();
+        return customerConverter.toDto(allCustomers);
     }
 
-    public Customer getCustomerById(Integer id) {
-        return customerRepository.findById(id).orElse(null);
+    public CustomerDTO getCustomerById(Integer id) {
+        Customer customerById = customerRepository.findById(id).orElse(null);
```

this is duplicated code, could be extracted to a private method

---

you could also throw a not found exception here